studiopress/genesis-custom-blocks
# Template and onboarding notices, bugfixes

Superseded by https://github.com/studiopress/genesis-custom-blocks/pull/55 (that's branched off this)

Updated December 22

#### Tickets
[GF-2514](https://wpengine.atlassian.net/browse/GF-2514): Clipboard copy block filename
[GF-2515](https://wpengine.atlassian.net/browse/GF-2515): Template file location notice
[GF-2516](https://wpengine.atlassian.net/browse/GF-2516): Onboarding notices
[GF-2502](https://wpengine.atlassian.net/browse/GF-2502): Dynamically show field settings

#### Changes
* Adds onboarding notices
* Adds template notices

#### Testing instructions
1. `composer install && npm install && npm run build`
1. No need to activate the Pro plugin for this PR's testing instructions
1. Move all of your GCB blocks to the trash: <img width="1189" alt="the-ones-here" src="https://user-images.githubusercontent.com/4063887/102915780-f8a71b80-4447-11eb-9aa6-72453ab7ced4.png">
1. Deactivate GCB
1. Activate GCB
1. Click here: <img width="1002" alt="click-here" src="https://user-images.githubusercontent.com/4063887/102915963-520f4a80-4448-11eb-999a-2717e9f42462.png">
1. Expected: ![some-notices-exp](https://user-images.githubusercontent.com/4063887/102916636-7cadd300-4449-11eb-8fed-167b46735d76.gif)
1. Publish the block
1. Expected: there's a different notice: ![the-bisf](https://user-images.githubusercontent.com/4063887/102917074-2b521380-444a-11eb-8087-02eb4ace12f9.gif)
1. Click the 'Title' field: <img width="1344" alt="title-field" src="https://user-images.githubusercontent.com/4063887/102919071-cac4d580-444d-11eb-8b63-941f8fc9d6bc.png">
1. Change the 'Field' to each of the 13 values (Text, Textarea, URL, etc...): 
<img width="1344" alt="title-field" src="https://user-images.githubusercontent.com/4063887/102919187-06f83600-444e-11eb-966b-3064760ee374.png">
1. Expected: there are usually different settings fields when you change the 'Field':
<img width="1344" alt="title-field" src="https://user-images.githubusercontent.com/4063887/102919617-c8af4680-444e-11eb-88e8-8070c6470b0e.gif">
